src/common/deflate.cpp

```diff
@@ -81,6 +81,63 @@ byte *decompressDeflate(const byte *data, size_t inputSize,
 	return decompressedData.release();
 }
 
+byte *decompressDeflate2(const byte *data, size_t inputSize, size_t &outputSize,
+						int windowBits, unsigned int frameSize) {
+	/* Initialize the zlib data stream for decompression with our input data.
+	 *
+	 * This ugly const cast is necessary because the zlib API wants a non-const
+	 * next_in pointer by default. Unless we define ZLIB_CONST, but that only
+	 * appeared in zlib 1.2.5.3. Not really worth bumping our required zlib
+	 * version for, IMHO. */
+
+	z_stream strm;
+	strm.zalloc   = Z_NULL;
+	strm.zfree    = Z_NULL;
+	strm.opaque   = Z_NULL;
+	strm.avail_in = inputSize;
+	strm.next_in  = const_cast<byte *>(data);
+
+	int zResult = inflateInit2(&strm, windowBits);
+	BOOST_SCOPE_EXIT( (&strm) ) {
+			inflateEnd(&strm);
+	} BOOST_SCOPE_EXIT_END
+
+	std::vector<byte *> buffers;
+	ScopedArray<byte> decompressedDataBuffer(new byte[frameSize]);
+
+	do {
+		// Set the output data pointer and size
+		strm.avail_out = frameSize;
+		strm.next_out = decompressedDataBuffer.get();
+
+		// Decompress. Z_FULL_FLUSH, because we want to decompress partwise.
+		zResult = inflate(&strm, Z_FULL_FLUSH);
+		if (zResult != Z_STREAM_END && zResult != Z_OK)
+			throw Exception("Failed to inflate: %s (%d)", zError(zResult), zResult);
+
+		buffers.push_back(new byte[frameSize]);
+		if (strm.total_out % frameSize)
+			std::memcpy(buffers[buffers.size()-1], decompressedDataBuffer.get(), strm.total_out % frameSize);
+		else
+			std::memcpy(buffers[buffers.size()-1], decompressedDataBuffer.get(), frameSize);
```

Comment:
Is that necessary? Can't you decompress directly into the `buffers` memory?

Reply:
The problem is, that i don't know at the beginning how big the buffer should be.

Reply:
But you know the max size, frameSize. So you decompress directly into frameSize and keep a separate vector how much of each buffer is filled

Reply:
Oh, i recognized, this is another  piece of code, i initially thought it is. Simply ignore my last comment.